lib/create-file.js

```diff
@@ -1,27 +1,25 @@
 'use strict';
 
 const templateFactory = require('./template-factory');
-const { writeFile } = require('../lib/lib-helpers');
+const { writeFile } = require('../lib/helpers');
 const path = require('path');
 
-const createComponent = (name, isDumb, createDir, overwriteFile, options) => {
-    let dirPath = path.normalize(path.resolve(name));
-    let file = `${dirPath}.js`;
-    let compName = path.basename(name);
-    let template = isDumb
-        ? templateFactory('DUMB_COMPONENT', compName)
-        : templateFactory('COMPONENT', compName, options);
+const createComponent = (fileName, filePath, options, configObj) => {
+    let fileObj = path.parse(filePath);
+    filePath = fileObj.ext ? filePath : `${filePath}.js`;
+    let template = options.dumb
+        ? templateFactory('DUMB_COMPONENT', fileName)
+        : templateFactory('COMPONENT', fileName, configObj);
 
-    writeFile(file, template, overwriteFile, createDir);
+    writeFile(filePath, template, options.overwrite, options.createDir);
```

Comment:
I think a lot of this is my own fault, but we may want to try and find some better names for variables here... 

What do you think about going forward, rather than writing comments to try and explain what something is or what it's doing, let's try to create really robust names that are self explanatory as to what that variable is referencing?  I'm just trying to look at it from the perspective of someone who didn't know the code at all.... so maybe we can try to keep that in mind going forward to be a little more verbose (and again, that's mostly my fault... I'm really bad about choosing very generic variable names, lol).

Reply:
lol, I'll edit these variable names and keep that in mind moving word.

Reply:
Yea, I'm gonna try to get better at that as well... I'm hoping at some point that more people will join in to contribute, so I need to do a better job of keeping that mind as I code.  Bad habits are hard to break, though, lol!